.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,102 @@
+---
+
+name: Build and Test
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+env:
+  IMAGE_PREFIX: "ghcr.io/${{ github.repository }}"
+  REGISTRY: 'ghcr.io'
+
+jobs:
+  build-test-push:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - workdir: microservice_a
+          - workdir: microservice_b
+          - workdir: microservice_c
+      fail-fast: false
```

Comment:
to continue building other images

Reply:
• Deploy the passing docker images to the image repository, but do NOT deploy the failing 
image. 

2/3 in that case